src/main/java/com/aws/greengrass/shadowmanager/sync/model/FullShadowSyncRequest.java

```diff
@@ -205,12 +215,16 @@ public void execute(SyncContext context) throws RetryableException, SkipSyncRequ
 
     /**
      * Check if this is the first time there is a sync for the thing's shadow by checking the last sync time.
+     * Also check if a shadow was updated after the last sync.
      *
-     * @param syncInformation The sync information for the thing's shadow.
+     * @param syncInformation  The sync information for the thing's shadow.
+     * @param shadowUpdateTime The time the shadow was updated.
      * @return true if this is the first time the shadow is being sync; Else false.
      */
-    private boolean isFirstSync(@NonNull SyncInformation syncInformation) {
-        return syncInformation.getLastSyncTime() == Instant.EPOCH.getEpochSecond();
+    private boolean isFirstSyncOrShadowUpdatedAfterSync(@NonNull SyncInformation syncInformation,
```

Comment:
isFirstSync Or CloudShadowUpdatedAfterSync?

Reply:
Not really. We do a similar check for local shadow as well.